workflow/workflow.go

```diff
@@ -360,37 +381,46 @@ func GetVersion(ctx Context, changeID string, minSupported, maxSupported Version
 // Channel.Get() or Future.Get(). Trying to do so in query handler code will fail the query and client will receive
 // QueryFailedError.
 // Example of workflow code that support query type "current_state":
-//  func MyWorkflow(ctx workflow.Context, input string) error {
-//    currentState := "started" // this could be any serializable struct
-//    err := workflow.SetQueryHandler(ctx, "current_state", func() (string, error) {
-//      return currentState, nil
-//    })
-//    if err != nil {
-//      currentState = "failed to register query handler"
-//      return err
-//    }
-//    // your normal workflow code begins here, and you update the currentState as the code makes progress.
-//    currentState = "waiting timer"
-//    err = NewTimer(ctx, time.Hour).Get(ctx, nil)
-//    if err != nil {
-//      currentState = "timer failed"
-//      return err
-//    }
-//
-//    currentState = "waiting activity"
-//    ctx = WithActivityOptions(ctx, myActivityOptions)
-//    err = ExecuteActivity(ctx, MyActivity, "my_input").Get(ctx, nil)
-//    if err != nil {
-//      currentState = "activity failed"
-//      return err
-//    }
-//    currentState = "done"
-//    return nil
-//  }
+//
+//	func MyWorkflow(ctx workflow.Context, input string) error {
+//	  currentState := "started" // this could be any serializable struct
+//	  err := workflow.SetQueryHandler(ctx, "current_state", func() (string, error) {
+//	    return currentState, nil
+//	  })
+//	  if err != nil {
+//	    currentState = "failed to register query handler"
+//	    return err
+//	  }
+//	  // your normal workflow code begins here, and you update the currentState as the code makes progress.
+//	  currentState = "waiting timer"
+//	  err = NewTimer(ctx, time.Hour).Get(ctx, nil)
+//	  if err != nil {
+//	    currentState = "timer failed"
+//	    return err
+//	  }
+//
+//	  currentState = "waiting activity"
+//	  ctx = WithActivityOptions(ctx, myActivityOptions)
+//	  err = ExecuteActivity(ctx, MyActivity, "my_input").Get(ctx, nil)
+//	  if err != nil {
+//	    currentState = "activity failed"
+//	    return err
+//	  }
+//	  currentState = "done"
+//	  return nil
+//	}
 func SetQueryHandler(ctx Context, queryType string, handler interface{}) error {
 	return internal.SetQueryHandler(ctx, queryType, handler)
 }
 
+func SetUpdateHandler(ctx Context, updateName string, handler interface{}) error {
+	return SetUpdateHandlerOpts(ctx, updateName, handler, UpdateHandlerOptions{})
+}
+
+func SetUpdateHandlerOpts(ctx Context, updateName string, handler interface{}, opts UpdateHandlerOptions) error {
```

Comment:
Question - about how far out are we from having an integration test or two for updates? Would totally support _only_ using SDK features if you'd like though that often requires release coordination so I understand just an integration test or two in here.

Reply:
Server hasn't stabilized to the point where putting integration tests in this repo would be a good idea. There are sdk-feature tests but they can't land in main until this PR lands since they depend on public API stuff.